### DISCUSSION TITLE
do not update field when post to livewire is returned

### DISCUSSION THREAD

**Discussion by iateadonut** [livewire/livewire#5419] [1. Help]
We have a mask that is dynamically added:

```
    var masks = {
        phone: '(000) 000-0000'
    }

    function setupMasks() {
        var elms = document.querySelectorAll('[data-mask]');
        var m = [];
        for(var i =0; i < elms.length; i++) (function(i) {
            var elm = elms[i];
            var mask  = masks[ elm.getAttribute('data-mask') ];
            m[i] = new IMask(elm, typeof mask == 'object' ? mask : {mask: mask});
            m[i].updateOptions({lazy: true});
            
            elm.addEventListener('focus', function() {
                m[i].updateOptions({lazy: false});
            })

            elm.addEventListener('blur', function() {
                if(this.value == '(___) ___-____' ||
                this.value == '(' ||
                this.value == '' ) {
                    this.value = '';
                    m[i].updateOptions({lazy: true});
                }
            })

        })(i);
    }
```

We only want to turn "lazy" on if there is no input value.  This code is working fine, as far as the js goes.

This is the input field

```
<input type="text" 
    wire:ignore=""
   data-mask="phone"
   wire:model.lazy="cell_phone"
   name="cell_phone"
   class="form-control"
   phone="phone"
   placeholder="Cell Phone #"
   value="" >
```

The problem is that when it is POSTed to livewire, without user input, \(\_\_\_) \_\_\_-____ is sent to livewire as the value, and when the POST responds, it updates the input to be \(\_\_\_) \_\_\_-____.  I am using wire:ignore, so I would expect it not to update based on the response.

**Comment by iateadonut** [comment]
I ended up doing this within the component:

```
    public function updated($key, $value) {
        if(strpos($key, 'phone') !== false) {
            $this->$key = $value;
            if ( in_array($value, ['(', '(___) ___-____']) ) {
                $this->$key = '';
            }
            // IF LAZY LOADING ON NON-EMPTY BLUR, USE THIS:
            // $this->$key = str_replace('_', '' , str_replace(') _', '', str_replace('-_', '', $this->$key)));
        }
    }
```
but I think there is a more correct way.

**Comment by joshhanley** [comment]
@iateadonut have you tried using Alpine's mask plugin instead of writing it yourself? https://alpinejs.dev/plugins/mask

**Reply by iateadonut** [reply]
Will do, thanks!

That being the proper solution aside, is there a way to prevent livewire from updating the value of a specific input field that is tied to a model?

**Reply by joshhanley** [reply]
@iateadonut no, you can defer the update using `.defer` but otherwise `wire:model` will still run.

Or remove `wire:model` completely and use Alpine to control it either using `@entangle` or updating Livewire's properties manually using `$wire`. This is what I would do.

**Reply by iateadonut** [reply]
Thank you.
